call/contract.go

```diff
@@ -194,68 +200,131 @@ func parseNewMethod(signature string) Method {
+		newMethod.Outputs = parseArguments(outputPath, "output")
 	} else {
 		singleReturnPaths := strings.Split(signature, ")")
 		paramsPaths := strings.Split(singleReturnPaths[0], "(")
-		params := parseParamsPath(paramsPaths[1])
+		params := strings.Split(paramsPaths[1], ",")
 
 		if len(params) > 0 {
-			for _, inParam := range params {
+			for i, inParam := range params {
 				if inParam != "" {
 					newMethod.Inputs = append(newMethod.Inputs, Argument{
-						Key:          "",
-						Type:         strings.TrimSpace(inParam),
-						InternalType: strings.TrimSpace(inParam),
+						Component: Component{
+							Name:         fmt.Sprintf("input%d", i),
+							Type:         strings.TrimSpace(inParam),
+							InternalType: strings.TrimSpace(inParam),
+						},
 					})
 				}
 			}
 		}
 
 		returnType := strings.TrimSpace(singleReturnPaths[1])
 		newMethod.Outputs = append(newMethod.Outputs, Argument{
-			Key:          "",
-			Type:         returnType,
-			InternalType: returnType,
+			Component: Component{
+				Name:         "output",
+				Type:         returnType,
+				InternalType: returnType,
+			},
 		})
 	}
 	return newMethod
 }
 
-func parseParamsPath(paramsPath string) []string {
-	params := strings.Split(paramsPath, ",")
-	return params
-}
-
 func repackAbi(methods []Method) (abi.ABI, error) {
 	abiString, err := json.Marshal(methods)
 	if err != nil {
 		return abi.ABI{}, err
 	}
 	return abi.JSON(strings.NewReader(string(abiString)))
 }
+
+func parseArguments(path, nameFormat string) []Argument {
+	result := []Argument{}
+	if path == "" {
+		return result
+	}
+	arguments := parsePath(path)
+	for i, inArgument := range arguments {
+		name := fmt.Sprintf(nameFormat+"%d", i)
+		argumentType, isTuple := getArgumentType(inArgument)
+		argument := Argument{
+			Component: Component{
+				Name:         name,
+				Type:         argumentType,
+				InternalType: argumentType,
+			},
+			Components: make([]Component, 0),
+		}
+
+		if isTuple {
+			s := strings.ReplaceAll(inArgument, "(", "")
+			s = strings.ReplaceAll(s, ")", "")
+			s = strings.ReplaceAll(s, "[]", "")
+			components := strings.Split(s, ",")
+			for j, component := range components {
+				argument.Components = append(argument.Components,
+					Component{
+						Name:         name + fmt.Sprintf("component%d", j),
+						Type:         strings.TrimSpace(component),
+						InternalType: strings.TrimSpace(component),
+					},
+				)
+			}
+		}
+		result = append(result, argument)
+	}
+	return result
+}
+
+func getArgumentType(inArgument string) (string, bool) {
+	if strings.Contains(inArgument, "(") {
+		if strings.Contains(inArgument, "[]") {
+			return "tuple[]", true
+		}
+		return "tuple", true
+	}
+	return strings.TrimSpace(inArgument), false
+}
+
+func parsePath(path string) []string {
+	arguments := []string{}
+	i := 0
+	n := len(path)
+	for {
+		accumulateString := ""
+		char := string(path[i])
+		if char == "(" {
```

Comment:
I'm quite concerning with this parser, is it general enough for all of the function signature case, can you test it with the method below?
`claimableRewards(address)((address,uint256)[], (address,address)[])`